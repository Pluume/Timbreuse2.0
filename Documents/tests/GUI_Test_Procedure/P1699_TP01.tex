\documentclass[10pt,a4paper,onecolumn]{article}
\usepackage[top=30pt,bottom=30pt,left=48pt,right=46pt]{geometry}
\usepackage{a4wide}
\usepackage{titling}
\setlength{\droptitle}{-5em}
\usepackage[version=3]{mhchem}
\usepackage{siunitx}
\usepackage{graphicx}
\usepackage{amsmath}
\usepackage[utf8]{inputenc}
\usepackage[english,francais]{babel}
\usepackage[backend=bibtex]{biblatex}
\usepackage[pdftex]{hyperref}
\setlength\parindent{0pt}
\usepackage{xcolor}
\hypersetup{colorlinks,urlcolor=blue}
\usepackage{fancyhdr}
\usepackage{lastpage}
\pagestyle{fancy}
\fancyhf{}
\chead{Francis Le Roy}
\lfoot{P1699/MCT}
\cfoot{\today}
\rfoot{Page \thepage \hspace{1pt} of \pageref{LastPage}}
% \bibliography{bib}
\renewcommand{\labelenumi}{\alph{enumi}.}



\title{Procédure de test P1699\_TP01}

\author{Francis \textsc{Le Roy}}

\date{\today}

\begin{document}

\maketitle
\thispagestyle{fancy}

\begin{center}
\begin{tabular}{l r}
Date de réalisation : & 13 Mai, 2017 \\
Projet : & P1699
\end{tabular}
\end{center}

\section{Introduction}
Les tests effectués sur l'interface graphique de la timbreuse ont des procédures et des résultats similaires. De ce fait, toutes les fonctionnalités peuvent être testées grâce à la méthode suivante.
\section{Organisation visuelle}
Les méthodes de test sont similaires entre tous les tests concernant les panneaux des administrateurs, des enseignants et des élèves. Sur ces panneaux, une interface parente que l'on nommera «Conteneur», contient des liens vers les différentes pages du panel correspondant. 
Chaque lien redirige la page enfant que l'on nommera «Hôte».\\

\section{Méthode de tests}
Pour pouvoir tester les éléments de l'interface graphique, il est nécessaire de se connecter sur le logiciel de la timbreuse avec un identifiant et un mot de passe valides pour pouvoir accéder au panneau correspondant au rang de l'utilisateur connecté. Un fois sur son panneau, de nombreux contrôles, sous la forme de boutons, de calendriers, de boîtes de texte ou de boîtes à cocher, sont disponibles. \\
Pour tester la fonctionnalité étudiée, l'utilisateur doit naviguer dans le logiciel de la timbreuse jusqu'à pouvoir accéder au contrôle qui va activer la fonction testée.

\section{Interprétation des résultats obtenus}
\subsection{Succès}
On identifie un succès généralement par un affichage des informations recherchées lors de l'exécution de la fonction étudiée, ou bien par une boîte de texte de fond vert au sommet de l'application avec un message décrivant le résultat de la fonction. Cette boîte de texte disparaît au bout de 2 secondes.
\subsection{Erreurs}
On identifie généralement une erreur par une absence de réaction après avoir déclenché la fonction. Si la fonction c'est exécutée correctement, mais qu'une erreur connue a été détectée, alors une boîte de texte de fond rouge apparaîtra pendant 15 secondes avec un court message décrivant l'erreur.
\\
Pour avoir plus d'informations sur une erreur, on doit, lorsque la fenêtre du logiciel de la timbreuse est sélectionnée, faire la combinaison de touche "Ctrl+I". Un panneau s'ouvrira sur la droite avec un champs de texte nommé "console". Des messages d'erreurs pourraient être affichés ici si l'erreur provient du processus de rendu graphique. Si l'erreur venait du processus principal, alors l'erreur serait affichée dans le fichier de log, situé à la racine du logiciel.
\section{Conclusion}
Si la fonctionnalité a été un succès, on peut donc définir cette fonctionnalité comme opérationnelle. 
% \printbibliography

\end{document}

