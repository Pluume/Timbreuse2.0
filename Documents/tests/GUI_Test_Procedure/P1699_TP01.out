\BOOKMARK [1][-]{section.1}{Introduction}{}% 1
\BOOKMARK [1][-]{section.2}{Organisation visuelle}{}% 2
\BOOKMARK [1][-]{section.3}{M\351thode de tests}{}% 3
\BOOKMARK [1][-]{section.4}{Interpr\351tation des r\351sultats obtenus}{}% 4
\BOOKMARK [2][-]{subsection.4.1}{Succ\350s}{section.4}% 5
\BOOKMARK [2][-]{subsection.4.2}{Erreurs}{section.4}% 6
\BOOKMARK [1][-]{section.5}{Conclusion}{}% 7
